src/common/src/types/mod.rs

```diff
@@ -322,6 +322,15 @@ for_all_scalar_variants! { scalar_impl_enum }
 pub type Datum = Option<ScalarImpl>;
 pub type DatumRef<'a> = Option<ScalarRefImpl<'a>>;
 
+pub fn get_data_type_from_datum(datum: &Datum) -> Result<DataType> {
+    match datum {
+        None => Err(internal_error(
```

Comment:
I think the data type of a literal should be unknown until it's inferred by some means, think of such case:

```
postgres=# create type foo as (v1 int, v2 int);                                                                                           CREATE TYPE
postgres=# create table t (v1 foo);
CREATE TABLE
postgres=# SELECT * FROM t WHERE v1 > (1, null);
 v1
----
(0 rows)
```

How can we determine the type of `(1, null)` until it's inferred from v1? (given the rule that v1 and `(1, null)` must be the same type).

Reply:
I see.., will fix in follow-up issue.